packages/ddp-server/livedata_server.js

```diff
@@ -1601,12 +1567,57 @@ _.extend(Server.prototype, {
     // cursor observe callbacks have fired when your callback is invoked. (We
     // can change this if there's a real use case.)
     if (callback) {
-      callback(exception, result);
-      return undefined;
+      promise.then(
+        result => callback(undefined, result),
+        exception => callback(exception)
+      );
+    } else {
+      return promise.await();
     }
-    if (exception)
-      throw exception;
-    return result;
+  },
+
+  // @param options {Optional Object}
+  applyAsync: function (name, args, options) {
```

Comment:
I think I like that as a general rule: suffix all `Promise`-returning functions with -Promise, instead of -Async.